sdk/eventhub/azure-eventhub/samples/async_samples/recv_process_individual_checkpoint_by_time.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import asyncio
+import os
+import logging
+import time
+from azure.eventhub.aio import EventHubConsumerClient
+from azure.eventhub.extensions.checkpointstoreblobaio import BlobCheckpointStore
+
+CONNECTION_STR = os.environ["EVENT_HUB_CONN_STR"]
+STORAGE_CONNECTION_STR = os.environ["AZURE_STORAGE_CONN_STR"]
+BLOB_CONTAINER_NAME = "your-blob-container-name"  # Please make sure the blob container resource exists.
+
+logging.basicConfig(
+    level=logging.INFO,
+    format='%(threadName)s | %(asctime)s | %(name)-12s | %(levelname)-8s %(message)s', datefmt='%m-%d %H:%M:%S',)
+log = logging.getLogger(__name__)
+
+MAX_UPLOAD_DOC_TASK_COUNT = 100
+upload_doc_task_count = 0
+
+
+def reduce_task_count(fut):
+    global upload_doc_task_count
```

Comment:
I would like to avoid using `global` in our examples. It is generally something that indicate that there is a structural problem in the code. Not always, but often enough.

Reply:
Removed it after changing to use a semaphore